src/lib/UniversalDiscovery/UniversalDiscoveryProvider.php

```diff
@@ -248,7 +249,14 @@ public function getSubitemLocations(
         return [
             'locations' => array_map(
                 function (SearchHit $searchHit) {
-                    return $this->getRestFormat($searchHit->valueObject);
+                    /** @var \eZ\Publish\API\Repository\Values\Content\Location $location */
+                    $location = $searchHit->valueObject;
+                    $restLocation = new RestLocation(
+                        $location,
+                        $this->locationService->getLocationChildCount($location)
```

Comment:
Have you checked whether necessity of calling `SearchService` for children count in a loop isn't more expensive in the end?

Reply:
It definitely isn't as expensive as fetching all fields in a given content. perhaps I could just put 0 there as this information is not necessary at all 🤔

Reply:
AFAICS there is a dedicated visitor exposing `childCount`, so not really, ref: https://github.com/ezsystems/ezplatform-rest/blob/1.3/src/lib/Server/Output/ValueObjectVisitor/RestLocation.php#L89.

Reply:
Yes, it will be 0 but why would we need the correct value there? It's not used in subitems list in UDW anywhere.

Reply:
Besides, this call was made also in the previous solution.

Reply:
It seems so, yes. I am up for keeping the API call instead of `0` then, performance-wise we should be good here.

Reply:
I would say this is arguable, UDW doesn't have to know how many children the current node has, we are not changing it globally but only for the UDW action. Calling it 50 times (as this is the default limit) as you have noticed can be respectfully damaging from the performance side.

Reply:
If we decide to go this way, we should at least make sure that this property isn't visited in `ibexa/rest` then as it would show wrong data. Not sure if we aren't breaking the BC policy though with that approach.

Reply:
I don't think I follow. In other places/calls the value of `childCount` would be properly set. We are changing it locally only for the single UDW action.

Reply:
Indeed, it seems this is only used internally, so saving unnecessary `SearchService` calls seems to be a way to go. +1 from my side.